Makefile

```diff
@@ -63,12 +82,37 @@ cross-local:
 	go build -o "bin/dnet-$$GOOS-$$GOARCH" ./cmd/dnet
 	go build -o "bin/docker-proxy-$$GOOS-$$GOARCH" ./cmd/proxy
 
+# Rebuild protocol buffers.
+# These may need to be rebuilt after vendoring updates, so .pb.go files are declared .PHONY so they are always rebuilt.
+PROTO_FILES=$(shell find . -path ./vendor -prune -o -name \*.proto -print)
+PB_FILES=$(PROTO_FILES:.proto=.pb.go)
+
+# Pattern rule for protoc.   If PROTOC_CHECK is defined, it checks
+# whether the generated files are up to date and fails if they are not
+%.pb.go: %.proto
+	@if [ ${PROTOC_CHECK} ]; then \
+	protoc -I=. -I=/go/src -I=/go/src/github.com/gogo/protobuf -I=/go/src/github.com/gogo/protobuf/protobuf --gogo_out=/tmp $< ; \
+	diff -q $@ /tmp/$@ >/dev/null || (echo "👹 $@ is out of date; please run 'make protobuf' and check in updates" && exit 1) ; \
```

Comment:
what if we pass the `--gogo_out` path directly as an ENV variable like we do for dockerbuildargs (https://github.com/docker/libnetwork/pull/2217/files#diff-1d37e48f9ceff6d8030570cd36286a61L9)
that way we don't need PROTOC_CHECK, locally you will always want to regenerate them eventually.

Reply:
or you can also use the notion of the CIRCLE_CI env variable that is already passed and set at the top of the makefile the output path so no need to specify other env variables

Reply:
I would prefer for `make check` to work in the same way whether you run it in CircleCI or on your own machine, so I can't use the CIRCLE_CI variable.

Reply:
I also tried passing `--gogo_out` path as a variable, but then you still have to decide whether or not to run `diff` so it doesn't help much.